Render track artist when they don't match album artist

Nemo157 opened the pull request:
An example of an album where every track has a different artist:

<img width="803" alt="image" src="https://user-images.githubusercontent.com/81079/87223103-64a23580-c37a-11ea-8fb4-06cc8e649962.png">

and one where only a few tracks are featuring other artists:

<img width="802" alt="image" src="https://user-images.githubusercontent.com/81079/87223099-56ecb000-c37a-11ea-9daf-770334f61f7e.png">

Nemo157 commented:
Is it possible to see what is failing in CI without signing up for a bitrise account?

danbee commented:
@Nemo157 Probably not, but you can sign into Bitrise with your GitHub account

Nemo157 commented:
🤷 after signing up I just get a 404 page when I click on the details.

danbee commented:
Here's the build output:

```
⚠️  /Users/vagrant/git/libmpdclient/src/socket.h:32:2: missing submodule 'mpdclient.socket' [-Wincomplete-umbrella]
#include <mpd/socket.h>
 ^       ~
⚠️  /Users/vagrant/git/libmpdclient/src/sync.c:32:2: missing submodule 'mpdclient.async' [-Wincomplete-umbrella]
#include <mpd/async.h>
 ^       ~
▸ Compiling sticker.c
▸ Compiling status.c
⚠️  /Users/vagrant/git/libmpdclient/src/status.c:231:27: implicit conversion loses integer precision: 'unsigned long' to 'unsigned int' [-Wshorten-64-to-32]
                status->queue_version = strtoul(pair->value, NULL, 10);
                                      ~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
⚠️  /Users/vagrant/git/libmpdclient/src/status.c:249:26: implicit conversion loses integer precision: 'unsigned long' to 'unsigned int' [-Wshorten-64-to-32]
                status->elapsed_time = strtoul(pair->value, &endptr, 10);
                                     ~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
⚠️  /Users/vagrant/git/libmpdclient/src/status.c:251:25: implicit conversion loses integer precision: 'unsigned long' to 'unsigned int' [-Wshorten-64-to-32]
                        status->total_time = strtoul(endptr + 1, NULL, 10);
                                           ~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
⚠️  /Users/vagrant/git/libmpdclient/src/status.c:258:58: implicit conversion loses integer precision: 'unsigned long' to 'unsigned int' [-Wshorten-64-to-32]
                status->elapsed_ms = strtoul(pair->value, &endptr, 10) * 1000;
                                   ~ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
▸ Compiling stats.c
▸ Compiling song.c
▸ Compiling socket.c
⚠️  /Users/vagrant/git/libmpdclient/src/socket.h:32:2: missing submodule 'mpdclient.socket' [-Wincomplete-umbrella]
#include <mpd/socket.h>
 ^       ~
⚠️  /Users/vagrant/git/libmpdclient/src/fd_util.h:39:2: missing submodule 'mpdclient.socket' [-Wincomplete-umbrella]
#include <mpd/socket.h>
 ^       ~
⚠️  /Users/vagrant/git/libmpdclient/src/socket.c:155:45: implicit conversion loses integer precision: 'const size_t' (aka 'const unsigned long') to 'socklen_t' (aka 'unsigned int') [-Wshorten-64-to-32]
                ret = connect(fd, address->addr, address->addrlen);
                      ~~~~~~~                    ~~~~~~~~~^~~~~~~
▸ Compiling settings.c
▸ Compiling send.c
▸ Compiling search.c
▸ Compiling run.c
▸ Compiling rplaylist.c
▸ Compiling response.c
▸ Compiling resolver.c
▸ Compiling replay_gain.c
▸ Compiling recv.c
⚠️  /Users/vagrant/git/libmpdclient/src/recv.c:31:2: missing submodule 'mpdclient.parser' [-Wincomplete-umbrella]
#include <mpd/parser.h>
 ^       ~
⚠️  /Users/vagrant/git/libmpdclient/src/iasync.h:32:2: missing submodule 'mpdclient.async' [-Wincomplete-umbrella]
#include <mpd/async.h>
 ^       ~
▸ Compiling rdirectory.c
▸ Compiling quote.c
▸ Compiling queue.c
▸ Compiling playlist.c
▸ Compiling player.c
▸ Compiling password.c
▸ Compiling partition.c
▸ Compiling parser.c
⚠️  /Users/vagrant/git/libmpdclient/src/parser.c:116:28: implicit conversion loses integer precision: 'long' to 'enum mpd_server_error' [-Wshorten-64-to-32]
                parser->u.error.server = strtol(p + 1, &p, 10);
 ^       ~
⚠️  /Users/vagrant/git/libmpdclient/src/parser.c:118:25: implicit conversion loses integer precision: 'long' to 'unsigned int' [-Wshorten-64-to-32]
                        parser->u.error.at = strtol(p + 1, &p, 10);
                                       ~ ^~~~~~~~~~~~~~~~~~~~~
▸ Compiling output.c
▸ Compiling neighbor.c
▸ Compiling mount.c
▸ Compiling mixer.c
▸ Compiling message.c
▸ Compiling list.c
▸ Compiling kvlist.c
▸ Compiling iso8601.c
⚠️  /Users/vagrant/git/libmpdclient/src/iso8601.c:94:9: implicit conversion loses integer precision: 'unsigned long' to 'unsigned int' [-Wshorten-64-to-32]
        year = strtoul(input, &endptr, 10);
                                           ~ ^~~~~~~~~~~~~~~~~~~~~
⚠️  /Users/vagrant/git/libmpdclient/src/iso8601.c:100:10: implicit conversion loses integer precision: 'unsigned long' to 'unsigned int' [-Wshorten-64-to-32]
        month = strtoul(input, &endptr, 10);
             ~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~
⚠️  /Users/vagrant/git/libmpdclient/src/iso8601.c:105:8: implicit conversion loses integer precision: 'unsigned long' to 'unsigned int' [-Wshorten-64-to-32]
        day = strtoul(input, &endptr, 10);
              ~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~
⚠️  /Users/vagrant/git/libmpdclient/src/iso8601.c:110:9: implicit conversion loses integer precision: 'unsigned long' to 'unsigned int' [-Wshorten-64-to-32]
        hour = strtoul(input, &endptr, 10);
            ~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~
⚠️  /Users/vagrant/git/libmpdclient/src/iso8601.c:115:11: implicit conversion loses integer precision: 'unsigned long' to 'unsigned int' [-Wshorten-64-to-32]
        minute = strtoul(input, &endptr, 10);
             ~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~
⚠️  /Users/vagrant/git/libmpdclient/src/iso8601.c:120:11: implicit conversion loses integer precision: 'unsigned long' to 'unsigned int' [-Wshorten-64-to-32]
        second = strtoul(input, &endptr, 10);
               ~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~
▸ Compiling ierror.c
⚠️  /Users/vagrant/git/libmpdclient/src/socket.h:32:2: missing submodule 'mpdclient.socket' [-Wincomplete-umbrella]
#include <mpd/socket.h>
               ~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~
▸ Compiling idle.c
▸ Compiling fingerprint.c
▸ Compiling fd_util.c
⚠️  /Users/vagrant/git/libmpdclient/src/fd_util.h:39:2: missing submodule 'mpdclient.socket' [-Wincomplete-umbrella]
#include <mpd/socket.h>
 ^       ~
▸ Compiling error.c
▸ Compiling entity.c
▸ Compiling directory.c
▸ Compiling database.c
▸ Compiling cstatus.c
▸ Compiling cstats.c
▸ Compiling cplaylist.c
▸ Compiling cpartition.c
▸ Compiling coutput.c
▸ Compiling connection.c
⚠️  /Users/vagrant/git/libmpdclient/src/connection.c:35:2: missing submodule 'mpdclient.async' [-Wincomplete-umbrella]
#include <mpd/async.h>
 ^       ~
⚠️  /Users/vagrant/git/libmpdclient/src/connection.c:36:2: missing submodule 'mpdclient.parser' [-Wincomplete-umbrella]
#include <mpd/parser.h>
 ^       ~
⚠️  /Users/vagrant/git/libmpdclient/src/connection.c:38:2: missing submodule 'mpdclient.socket' [-Wincomplete-umbrella]
#include <mpd/socket.h>
 ^       ~
⚠️  /Users/vagrant/git/libmpdclient/src/socket.h:32:2: missing submodule 'mpdclient.socket' [-Wincomplete-umbrella]
#include <mpd/socket.h>
 ^       ~
⚠️  /Users/vagrant/git/libmpdclient/src/iasync.h:32:2: missing submodule 'mpdclient.async' [-Wincomplete-umbrella]
#include <mpd/async.h>
 ^       ~
⚠️  /Users/vagrant/git/libmpdclient/src/connection.c:67:27: implicit conversion loses integer precision: 'unsigned long' to 'unsigned int' [-Wshorten-64-to-32]
        connection->version[0] = strtoul(tmp, &test, 10);
 ^       ~
⚠️  /Users/vagrant/git/libmpdclient/src/connection.c:76:28: implicit conversion loses integer precision: 'unsigned long' to 'unsigned int' [-Wshorten-64-to-32]
                connection->version[1] = strtoul(test + 1, &test, 10);
                               ~ ^~~~~~~~~~~~~~~~~~~~~~~
⚠️  /Users/vagrant/git/libmpdclient/src/connection.c:78:29: implicit conversion loses integer precision: 'unsigned long' to 'unsigned int' [-Wshorten-64-to-32]
                        connection->version[2] = strtoul(test + 1, &test, 10);
                                       ~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
▸ Compiling cneighbor.c
▸ Compiling cmount.c
▸ Compiling cmessage.c
▸ Compiling capabilities.c
▸ Compiling audio_format.c
⚠️  /Users/vagrant/git/libmpdclient/src/audio_format.c:51:44: implicit conversion loses integer precision: 'unsigned long' to 'uint32_t' (aka 'unsigned int') [-Wshorten-64-to-32]
                        audio_format->sample_rate = dsd * 44100 / 8;
                                               ~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
⚠️  /Users/vagrant/git/libmpdclient/src/audio_format.c:60:30: implicit conversion loses integer precision: 'unsigned long' to 'uint32_t' (aka 'unsigned int') [-Wshorten-64-to-32]
        audio_format->sample_rate = strtoul(p, &endptr, 10);
                                                  ~ ~~~~~~~~~~~~^~~
▸ Compiling async.c
⚠️  /Users/vagrant/git/libmpdclient/src/iasync.h:32:2: missing submodule 'mpdclient.async' [-Wincomplete-umbrella]
#include <mpd/async.h>
                                  ~ ^~~~~~~~~~~~~~~~~~~~~~~
⚠️  /Users/vagrant/git/libmpdclient/src/socket.h:32:2: missing submodule 'mpdclient.socket' [-Wincomplete-umbrella]
#include <mpd/socket.h>
 ^       ~
⚠️  /Users/vagrant/git/libmpdclient/src/async.c:35:2: missing submodule 'mpdclient.socket' [-Wincomplete-umbrella]
#include <mpd/socket.h>
 ^       ~
▸ Compiling MPDClientWrapper.c
▸ Linking Persephone
⚠️  ld: directory not found for option '-L/Users/vagrant/git/Shared/libmpdclient/output'
▸ Copying /Users/vagrant/git/libmpdclient/vapi/libmpdclient.vapi
▸ Copying /Users/vagrant/git/libmpdclient/include/mpd/version.h.in
▸ Copying /Users/vagrant/git/libmpdclient/build/cross/win64.txt
▸ Copying /Users/vagrant/git/libmpdclient/build/cross/win32.txt
▸ Copying /Users/vagrant/git/libmpdclient/build/configure.py
▸ Copying /Users/vagrant/git/libmpdclient/build/build.sh
▸ Copying /Users/vagrant/git/README.md
▸ Copying /Users/vagrant/git/Mac/CHANGELOG.md
▸ Copying /Users/vagrant/git/LICENSE.md
▸ Copying /Users/vagrant/git/Brewfile
▸ Compiling VolumeControlView.xib
▸ Compiling DraggedSongView.xib
▸ Compiling AlbumDetailView.xib
▸ Compiling AlbumViewItem.xib
▸ Compiling Main.storyboard
▸ Processing Info.plist
▸ Running script 'Run Script'
▸ Touching Persephone.app (in target 'Persephone' from project 'Persephone')
▸ Build Succeeded
▸ Running script 'Run Script'
▸ Processing Info.plist
▸ Compiling PersephoneUITests.swift
▸ Processing Info.plist
▸ Processing Info.plist
▸ Compiling PersephoneTests.swift
⚠️  /Users/vagrant/git/MacTests/PersephoneTests.swift:10:18: implicit import of bridging header 'Persephone-Bridging-Header.h' via module 'Persephone' is deprecated and will be removed in a later version of Swift
@testable import Persephone
 ^       ~
▸ Linking PersephoneUITests
▸ Linking PersephoneTests
⚠️  ld: directory not found for option '-L/Users/vagrant/git/Shared/libmpdclient/output'
▸ Copying /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/usr/lib/libXCTestSwiftSupport.dylib
▸ Copying /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/Library/PrivateFrameworks/XCTAutomationSupport.framework
▸ Copying /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/Library/Frameworks/XCTest.framework
▸ Touching PersephoneUITests.xctest (in target 'PersephoneUITests' from project 'Persephone')
▸ Copying /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/usr/lib/libXCTestSwiftSupport.dylib
▸ Copying /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/usr/lib/libXCTestBundleInject.dylib
▸ Copying /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/Library/PrivateFrameworks/XCTAutomationSupport.framework
▸ Copying /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/Library/Frameworks/XCTest.framework
▸ Touching PersephoneTests.xctest (in target 'PersephoneTests' from project 'Persephone')
All tests
Test Suite PersephoneTests.xctest started
PersephoneTests
    ✓ testExample (0.151 seconds)
    ◷ testPerformanceExample measured (0.000 seconds)
    ✓ testPerformanceExample (0.352 seconds)
Executed 2 tests, with 0 failures (0 unexpected) in 0.503 (0.514) seconds
Last lines of the Xcode's build log:
Test session results, code coverage, and logs:
	/Users/vagrant/Library/Developer/Xcode/DerivedData/Persephone-dycuailxxyvanlcleaxpollbinro/Logs/Test/Run-Persephone-2020.06.18_10-46-37-+0000.xcresult
Testing failed:
	PersephoneUITests:
		PersephoneUITests-Runner.app (5387) encountered an error (Unable to run UI Tests because Xcode Helper does not have permission to use Accessibility. To enable UI testing, go to the Security & Privacy pane in System Preferences, select the Privacy tab, then select Accessibility, and add Xcode Helper to the list of applications allowed to use Accessibility)
** TEST FAILED **
Testing started on 'My Mac'
Test failed, error: exit status 65
|                                                                              |
+---+---------------------------------------------------------------+----------+
| x | xcode-test-mac@1.4.1 (exit code: 1)                           | 3.3 min  |
+---+---------------------------------------------------------------+----------+
| Issue tracker: ...s://github.com/bitrise-steplib/steps-xcode-test-mac/issues |
| Source: https://github.com/bitrise-steplib/steps-xcode-test-mac              |
+---+---------------------------------------------------------------+----------+
                                          ▼
+------------------------------------------------------------------------------+

| (7) deploy-to-bitrise-io@1.9.7                                               |
+---+---------------------------------------------------------------+----------+
+---+---------------------------------------------------------------+----------+
| ✓ | deploy-to-bitrise-io@1.9.7                                    | 6.96 sec |
+---+---------------------------------------------------------------+----------+
| Update available: 1.9.7 -> 1.10.0                                            |
|                                                                              |
| Release notes are available below                                            |
| https://github.com/bitrise-steplib/steps-deploy-to-bitrise-io/releases       |
+---+---------------------------------------------------------------+----------+
                                          ▼
+------------------------------------------------------------------------------+

| (8) cache-push@2.2.3                                                         |
+---+---------------------------------------------------------------+----------+
+---+---------------------------------------------------------------+----------+
| - | cache-push@2.2.3                                              | 1.38 sec |
+---+---------------------------------------------------------------+----------+
| Update available: 2.2.3 -> 2.3.0                                             |
|                                                                              |
| Release notes are available below                                            |
| https://github.com/bitrise-steplib/steps-cache-push.git/releases             |
+---+---------------------------------------------------------------+----------+
                                          ▼
+------------------------------------------------------------------------------+

| (9) xcode-archive-mac@1.8.0                                                  |
+---+---------------------------------------------------------------+----------+
+---+---------------------------------------------------------------+----------+
| - | xcode-archive-mac@1.8.0                                       | 1.63 sec |
+---+---------------------------------------------------------------+----------+
                                          ▼
+------------------------------------------------------------------------------+

| (10) slack@3.1.3                                                             |
+------------------------------------------------------------------------------+
| id: slack                                                                    |
| version: 3.1.3                                                               |
| collection: https://github.com/bitrise-io/bitrise-steplib.git                |
| toolkit: go                                                                  |
| time: 2020-06-18T10:48:52Z                                                   |
+------------------------------------------------------------------------------+
|                                                                              |
Config:
- Debug: false
- WebhookURL: 
- APIToken: 
- Channel: notifications
- ChannelOnError: 
- Text: 
- TextOnError: 
- IconEmoji: 
- IconEmojiOnError: 
- IconURL: https://github.com/bitrise-io.png
- IconURLOnError: https://github.com/bitrise-io.png
- LinkNames: true
- Username: Bitrise
- UsernameOnError: Bitrise
- Color: #3bc3a3
- ColorOnError: #f0741f
- PreText: *Build Succeeded!*
- PreTextOnError: *Build Failed!*
- AuthorName: Wim Looman
- Title: Deal with album artist when dragging songs
- TitleOnError: 
- TitleLink: 
- Message: 
- MessageOnError: 
- ImageURL: 
- ImageURLOnError: 
- ThumbURL: 
- ThumbURLOnError: 
- Footer: Bitrise
- FooterIcon: https://github.com/bitrise-io.png?size=16
- TimeStamp: true
- Fields: App|Persephone
Branch|render-track-artist
Workflow|primary
- Buttons: View App|https://app.bitrise.io/app/903bd386fa92d3b4
View Build|https://app.bitrise.io/build/1564825689ec325b
Install Page|
Error: Both API Token and WebhookURL are empty. You need to provide one of them. If you want to use incoming webhooks provide the webhook url. If you want to use a bot to send a message provide the bot API token
WARN[10:48:55] Step (slack@3.1.3) failed, but was marked as skippable 
|                                                                              |
+---+---------------------------------------------------------------+----------+
| ! | slack@3.1.3 (exit code: 1)                                    | 4.01 sec |
+---+---------------------------------------------------------------+----------+
| Issue tracker: https://github.com/bitrise-io/steps-slack-message/issues      |
| Source: https://github.com/bitrise-io/steps-slack-message                    |
+---+---------------------------------------------------------------+----------+
+------------------------------------------------------------------------------+
|                               bitrise summary                                |
+---+---------------------------------------------------------------+----------+
|   | title                                                         | time (s) |
+---+---------------------------------------------------------------+----------+
| - | activate-ssh-key@4.0.5                                        | 1.60 sec |
+---+---------------------------------------------------------------+----------+
| ✓ | git-clone@4.0.18                                              | 14.76 sec|
+---+---------------------------------------------------------------+----------+
| ! | cache-pull@2.1.4 (exit code: 1)                               | 5.30 sec |
+---+---------------------------------------------------------------+----------+
| Issue tracker: https://github.com/bitrise-steplib/steps-cache-pull/issues    |
| Source: https://github.com/bitrise-steplib/steps-cache-pull.git              |
+---+---------------------------------------------------------------+----------+
| ✓ | Do anything with Script step                                  | 1.4 min  |
+---+---------------------------------------------------------------+----------+
| ✓ | certificate-and-profile-installer@1.10.3                      | 7.47 sec |
+---+---------------------------------------------------------------+----------+
| ✓ | recreate-user-schemes@1.0.2                                   | 3.65 sec |
+---+---------------------------------------------------------------+----------+
| x | xcode-test-mac@1.4.1 (exit code: 1)                           | 3.3 min  |
+---+---------------------------------------------------------------+----------+
| Issue tracker: ...s://github.com/bitrise-steplib/steps-xcode-test-mac/issues |
| Source: https://github.com/bitrise-steplib/steps-xcode-test-mac              |
+---+---------------------------------------------------------------+----------+
| ✓ | deploy-to-bitrise-io@1.9.7                                    | 6.96 sec |
+---+---------------------------------------------------------------+----------+
| Update available: 1.9.7 -> 1.10.0                                            |
|                                                                              |
| Release notes are available below                                            |
| https://github.com/bitrise-steplib/steps-deploy-to-bitrise-io/releases       |
+---+---------------------------------------------------------------+----------+
| - | cache-push@2.2.3                                              | 1.38 sec |
+---+---------------------------------------------------------------+----------+
| Update available: 2.2.3 -> 2.3.0                                             |
|                                                                              |
| Release notes are available below                                            |
| https://github.com/bitrise-steplib/steps-cache-push.git/releases             |
+---+---------------------------------------------------------------+----------+
| - | xcode-archive-mac@1.8.0                                       | 1.63 sec |
+---+---------------------------------------------------------------+----------+
| ! | slack@3.1.3 (exit code: 1)                                    | 4.01 sec |
+---+---------------------------------------------------------------+----------+
| Issue tracker: https://github.com/bitrise-io/steps-slack-message/issues      |
| Source: https://github.com/bitrise-io/steps-slack-message                    |
+---+---------------------------------------------------------------+----------+
| Total runtime: 5.4 min                                                       |
+------------------------------------------------------------------------------+
Submitting anonymized usage informations...
For more information visit:
https://github.com/bitrise-io/bitrise-plugins-analytics/blob/master/README.md
```

danbee commented:
I think the important bits are:

```
Test Suite PersephoneTests.xctest started
PersephoneTests
    ✓ testExample (0.151 seconds)
    ◷ testPerformanceExample measured (0.000 seconds)
    ✓ testPerformanceExample (0.352 seconds)
Executed 2 tests, with 0 failures (0 unexpected) in 0.503 (0.514) seconds
Last lines of the Xcode's build log:
Test session results, code coverage, and logs:
	/Users/vagrant/Library/Developer/Xcode/DerivedData/Persephone-dycuailxxyvanlcleaxpollbinro/Logs/Test/Run-Persephone-2020.06.18_10-46-37-+0000.xcresult
Testing failed:
	PersephoneUITests:
		PersephoneUITests-Runner.app (5387) encountered an error (Unable to run UI Tests because Xcode Helper does not have permission to use Accessibility. To enable UI testing, go to the Security & Privacy pane in System Preferences, select the Privacy tab, then select Accessibility, and add Xcode Helper to the list of applications allowed to use Accessibility)
** TEST FAILED **
Testing started on 'My Mac'
Test failed, error: exit status 65
```

and

```
Error: Both API Token and WebhookURL are empty. You need to provide one of them. If you want to use incoming webhooks provide the webhook url. If you want to use a bot to send a message provide the bot API token
WARN[10:48:55] Step (slack@3.1.3) failed, but was marked as skippable
```

These might not be your fault!

danbee commented:
@Nemo157 What do you think about mirroring the song queue format instead? Keeps things consistent and might make the code a little easier:

<img width="272" alt="image" src="https://user-images.githubusercontent.com/165531/85044869-3905a400-b15c-11ea-84ef-e478fba5ab98.png">

Nemo157 commented:
That seems like it could work well, I'll see if I can figure out how to do so.

Nemo157 commented:
Updated the screenshots above. I'm sure there must be a nicer way to have a disappearing secondary field, but this is the first time I'm really touching any macOS GUI code.

danbee commented:
Hi @Nemo157, sorry for the long delay. Recent events have taken a toll on the amount of time I've got for side projects.

I'm not keen on the differing cell heights in the second screenshot. Could this be an all or nothing approach? If all the tracks match the album artist then don't show any per track artist, but if *some* of the tracks have different artists then show the artist for *all* the tracks in the list?

danbee commented on Mac/Components/Browser/Album Detail/AlbumDetailView+NSTableViewDelegate.swift:
```diff
@@ -81,9 +81,9 @@ extension AlbumDetailView: NSTableViewDelegate {
-
-    cellView.textField?.stringValue = song.title
+      ) as! AlbumDetailSongTitleView
+    
```
Let's remove trailing spaces.

danbee commented:
I've noticed too that somewhere we lost Disc headings on albums.

Nemo157 commented:
Take your time, I definitely understand not being able to work on side projects.

I've fixed the disc ids disappearing, the rows were getting auto-sized down to 0px high, and updated to enable/disable the track artist for all rows together.

danbee commented:
@Nemo157 This looks good! I'll take a look at the build issues.

sharplet commented on Mac/Components/Browser/Album Detail/AlbumDetailSongTitleView.swift:
```diff
@@ -0,0 +1,29 @@
+  
+  func setShowArtist(_ show: Bool) {
+    if collapseArtist == nil {
+       collapseArtist = songArtist?.heightAnchor.constraint(equalToConstant: 0.0)
```
(Note: I'm more familiar with stack views in UIKit, so if they don't work the same way feel free to disregard!)

I wonder if putting `songTitle` and `songArtist` in a vertical stack view would simplify this code a little? It would be handy if setting `songArtist.isHidden = true` automatically collapsed the view, without having to manage constraints.

danbee replied:
@sharplet The vertical stack view approach works great!

sharplet commented:
@danbee Looks like those might be the stock tests from the Xcode template? You could probably delete them 🤷

danbee commented:
Released in [0.17.2-alpha](https://github.com/danbee/persephone/releases/tag/0.17.2-alpha). Thanks for your contribution @Nemo157!
